static/js/components/forms/RegisterDetailsForm.js

```diff
@@ -1,41 +1,296 @@
 // @flow
 import React from "react"
 import * as yup from "yup"
+import { __, includes, find, propEq } from "ramda"
 
-import { Formik, Field, Form, ErrorMessage } from "formik"
+import { Formik, Field, Form, ErrorMessage, FieldArray } from "formik"
+import type { Country } from "../../flow/authTypes"
+import FormError from "./FormError"
+
+const US_ALPHA_2 = "US"
+const CA_ALPHA_2 = "CA"
+
+const US_POSTAL_CODE_REGEX = /[0-9]{5}(-[0-9]{4}){0,1}/
+const CA_POSTAL_CODE_REGEX = /[0-9][A-Z][0-9] [A-Z][0-9][A-Z]/
+const COUNTRIES_REQUIRING_POSTAL_CODE = [US_ALPHA_2, CA_ALPHA_2]
+const COUNTRIES_REQUIRING_STATE = [US_ALPHA_2, CA_ALPHA_2]
+
+const ADDRESS_LINES_MAX = 5
 
 const detailsValidation = yup.object().shape({
-  name:     yup.string().required(),
+  name: yup
+    .string()
+    .label("Legal Name")
+    .required(),
   password: yup
     .string()
+    .label("Password")
     .required()
     .min(8)
+    .matches(/^(?=.*[0-9])(?=.*[a-zA-Z]).*$/, {
+      message: "Password must contain at least one letter and number"
+    }),
+  legal_address: yup.object().shape({
+    first_name: yup
+      .string()
+      .label("First Name")
+      .required(),
+    last_name: yup
+      .string()
+      .label("Last Name")
+      .required(),
+    city: yup
+      .string()
+      .label("City")
+      .required(),
+    street_address: yup
+      .array()
+      .label("Street address")
+      .of(yup.string().max(60))
+      .min(1)
+      .max(ADDRESS_LINES_MAX)
+      .compact()
+      .required(),
+    state_or_territory: yup
+      .mixed()
+      .label("State/Territory")
+      .when("country", {
+        is:   includes(__, COUNTRIES_REQUIRING_STATE),
+        then: yup
+          .string()
+          .required()
+          .matches(/[A-Z]{2}-[A-Z]{2,3}/)
```

Comment:
I would remove this. The values should always match this and it causes the error message `State/Territory must match the following: "/[A-Z]{2}-[A-Z]{2,3}/"` which will cause the user to get confused.